c7n/resources/s3.py

```diff
@@ -1416,35 +1417,26 @@ def process(self, buckets, event=None):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        enabled = self.data.get('enabled', False)
-        scope = self.data.get('scope', 'Any')
+        config = {}
         try:
             config = s3.get_public_access_block(
                 Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
         except ClientError as e:
             if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else in matches_filter func
-                config = None
+                # When there is a None config, set all to false to be symbolic
+                for key in self.keys:
+                    config[key] = False
             else:
                 raise
-        if self.matches_filter(config, enabled, scope):
-            return {"Name": bucket['Name'], "publicblocks": config}
-
-    def matches_filter(self, config, enabled, scope):
-        if config:
-            if scope == 'All':
-                return all(config.values()) if enabled else not any(config.values())
-            elif scope == 'Any':
-                return any(config.values()) if enabled else not all(config.values())
-            else:
-                return config[scope] if enabled else not config[scope]
+        if self.matches_filter(config):
+            return {"Name": bucket['Name'], "PublicAccessBlockConfiguration": config}
+
+    def matches_filter(self, config):
+        key_set = [key for key in self.keys if type(self.data.get(key)) is bool]
```

Comment:
this implementation feels a little odd, type check to remove a key, and using indirect set equality instead of just equality, although perhaps its just style, here's an alternate.

```python
data = dict(self.data)
data.pop('type')

if data and data != {k: config[k] for k in data}:
    return False
elif not data:
    # with no configuration, we default to matching if any setting is False
    return bool([True for k, v in config.items() if v is False])
```

Reply:
Thanks! Yes at this point it's more of a style thing but I will play with this alternate and see how it can be applied. I am sure there are some nuggets of wisdom in here for me to consume in the effort. Will try and update a bit later today after breakfast.